actions/TestTaker.php

```diff
@@ -226,10 +226,11 @@ public function editSubject()
             $this->setData('reload', true);
         }
 
-        $assignmentService = $this->getServiceLocator()->get(AssignmentService::SERVICE_ID);
+        $currentUser = $userService->getCurrentUser();
+        $groupManagerRole = new core_kernel_classes_Resource(self::GROUP_MANAGER_ROLE);
         if (
             common_ext_ExtensionsManager::singleton()->isEnabled('taoGroups') &&
-            get_class($assignmentService) == GroupAssignment::class
+            $userService->userHasRoles($currentUser, $groupManagerRole)
```

Comment:
Before we did not have role restriction for this. Can we make sure that expected roles by business will still be able to manage this? As you know, when you create a user the role `Group Manager` is not available for selection. 

So I would make sure that business provide a list of roles they assume that should still have access for this and make sure they have `http://www.tao.lu/Ontologies/TAOGroup.rdf#GroupsManagerRole` linked to them (if necessary).